yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm

```diff
@@ -0,0 +1,161 @@
+	
+	var/charge = PRETERNIS_LEVEL_FULL
+	var/eating_msg_cooldown = FALSE
+	var/emag_lvl = 0
+	var/power_drain = 0.7 //probably going to have to tweak this shit
+	var/tesliumtrip = FALSE
+	var/draining = FALSE
+	
+
+/datum/species/preternis/on_species_gain(mob/living/carbon/C, datum/species/old_species, pref_load)
+	. = ..()
+	for (var/V in C.bodyparts)
+		var/obj/item/bodypart/BP = V
+		BP.change_bodypart_status(ORGAN_ROBOTIC,FALSE,TRUE)
+		BP.burn_reduction = 0
+		BP.brute_reduction = 0
+		if(istype(BP,/obj/item/bodypart/chest) || istype(BP,/obj/item/bodypart/head))
+			continue
+		BP.max_damage = 35
+	C.grant_language(/datum/language/machine) //learn it once,learn it forever i guess,this isnt removed on species loss to prevent curators from forgetting machine language
+
+/datum/species/preternis/on_species_loss(mob/living/carbon/human/C, datum/species/new_species, pref_load)
+	. = ..()
+	for (var/V in C.bodyparts)
+		var/obj/item/bodypart/BP = V
+		BP.change_bodypart_status(ORGAN_ORGANIC,FALSE,TRUE)
+		BP.burn_reduction = initial(BP.burn_reduction)
+		BP.brute_reduction = initial(BP.brute_reduction)
+	C.clear_alert("preternis_emag") //this means a changeling can transform from and back to a preternis to clear the emag status but w/e i cant find a solution to not do that
+	C.clear_fullscreen("preternis_emag")
+	C.remove_movespeed_modifier("preternis_teslium")
+
+/datum/species/preternis/spec_emp_act(mob/living/carbon/human/H, severity)
+	. = ..()
+	switch(severity)
+		if(EMP_HEAVY)
+			H.adjustBruteLoss(20)
+			H.adjustFireLoss(20)
+			H.Paralyze(50)
+			charge *= 0.4
+			H.visible_message("<span class='danger'>Electricity ripples over [H]'s subdermal implants, smoking profusely.</span>", \
+							"<span class='userdanger'>A surge of searing pain erupts throughout your very being! As the pain subsides, a terrible sensation of emptiness is left in its wake.</span>")
+		if(EMP_LIGHT)
+			H.adjustBruteLoss(10)
+			H.adjustFireLoss(10)
+			H.Paralyze(20)
+			charge *= 0.6
+			H.visible_message("<span class='danger'>A faint fizzling emanates from [H].</span>", \
+							"<span class='userdanger'>A fit of twitching overtakes you as your subdermal implants convulse violently from the electromagnetic disruption. Your sustenance reserves have been partially depleted from the blast.</span>")
+
+/datum/species/preternis/spec_emag_act(mob/living/carbon/human/H, mob/user)
+	. = ..()
+	if(emag_lvl == 2)
+		return
+	emag_lvl = min(emag_lvl + 1,2)
+	playsound(H.loc, 'sound/machines/warning-buzzer.ogg', 50, 1, 1)
+	H.Paralyze(60)
+	switch(emag_lvl)
+		if(1)
+			H.adjustBrainLoss(50) //HALP AM DUMB
+			to_chat(H,"<span class='danger'>ALERT! MEMORY UNIT [rand(1,5)] FAILURE.NERVEOUS SYSTEM DAMAGE.</span>")
+		if(2)
+			H.overlay_fullscreen("preternis_emag", /obj/screen/fullscreen/high)
+			H.throw_alert("preternis_emag", /obj/screen/alert/high/preternis)
+			to_chat(H,"<span class='danger'>ALERT! OPTIC SENSORS FAILURE.VISION PROCESSOR COMPROMISED.</span>")
```

Comment:
More of a suggestion than a complaint, but I would recommend looking into the effects for cyberman conversion for their hallucinations https://github.com/yogstation13/yogstation/tree/master/code/game/gamemodes/cybermen